src/components/MoneyInput/index.tsx

```diff
@@ -0,0 +1,63 @@
+import { Input } from 'antd';
+import type { FC } from 'react';
+import { useEffect } from 'react';
+import React, { useState } from 'react';
+import type { FormInstance, ProFormFieldProps } from '@ant-design/pro-form';
+import { ProForm } from '@ant-design/pro-form';
+
+interface CurrencyInputProps extends ProFormFieldProps {
+  name: string;
+  form: FormInstance<any>;
+  onChange?: (cents: number) => void;
+}
+
+export const MoneyInput: FC<CurrencyInputProps> = ({ name, form, onChange, ...restProps }) => {
+  const [dollars, setDollars] = useState('');
+  const [cents, setCents] = useState(0);
```

Comment:
use one `useState` instead of two 
 
```suggestion
  const [state, setState] = useState<{dollars:string, cents:numer}>({dollars:'', cents:0}});
```